### PR TITLE
gh-97669: Create Tools/build/ directory

### PR DESCRIPTION
**Pull request by vstinner**
Create Tools/build/ directory. Move the following scripts from Tools/scripts/ to Tools/build/:

* check_extension_modules.py
* deepfreeze.py
* freeze_modules.py
* generate_global_objects.py
* generate_levenshtein_examples.py
* generate_opcode_h.py
* generate_re_casefix.py
* generate_sre_constants.py
* generate_stdlib_module_names.py
* generate_token.py
* parse_html5_entities.py
* smelly.py
* stable_abi.py
* umarshal.py
* update_file.py
* verify_ensurepip_wheels.py

* Issue: gh-97669

### CODE REVIEW

**Comment by vstinner** [comment]
Update: add freeze_modules.py.

**Comment by JulienPalard** [comment]
To avoid the documentation check issues, you'll have to rebase on top of main, sorry for the noise :(((

**Comment by vstinner** [comment]
PR rebased to get fixes for the Docs CI job.

File: Tools/build/parse_html5_entities.py [hunk 1]
```diff
@@ -69,7 +70,7 @@ def write_items(entities, file=sys.stdout):
     keys = sorted(entities.keys())
     keys = sorted(keys, key=str.lower)
     print(HTML5_SECTION_START, file=file)
-    print(f'# Generated by {sys.argv[0]!r}\n'
+    print(f'# Generated by {SCRIPT_NAME}\n'
           f'# from {ENTITIES_URL} and\n'
           f'# {PAGE_URL}.\n'
           f'# Map HTML5 named character references to the '
```
Comment (ezio-melotti) [inline Tools/build/parse_html5_entities.py:76]: This file is listed in `.github/CODEOWNERS`, so that should be updated too.

Comment (vstinner) [reply Tools/build/parse_html5_entities.py:76]: **Updated**. Oh, I used a code search in all files tracked by Git, but I missed this ``Tools/scripts/*html5`` line in ``.github/CODEOWNERS``

**Comment by vstinner** [comment]
Update: I added **verify_ensurepip_wheels.py**. Technically, it's not directly required to *build* Python, but it's used by the workflow (CI).

**Comment by vstinner** [comment]
@tiran @erlend-aasland: Currently, these scripts are installed on Linux and macOS. IMO they are useful outside Python itself. So I propose to move them into a dedicated directory.

**Review by erlend-aasland — APPROVED** [review]

**Comment by vstinner** [comment]
> erlend-aasland approved these changes 2 hours ago 

Thanks for the review.

**Pull request merged**